tests/models/whisper/test_modeling_whisper.py

```diff
@@ -270,7 +271,6 @@ def get_config(self):
             attention_dropout=self.attention_probs_dropout_prob,
             max_position_embeddings=self.max_position_embeddings,
             max_source_positions=self.max_source_positions,
-            max_target_positions=self.max_target_positions,
```

Comment:
Let's add it back

Reply:
Thanks for your feedbacks.
I applied your suggestions.
I also changed [modeling_whisper.py](https://github.com/huggingface/transformers/pull/33196/files/3056095a45d2fbbd8ab818a68dd1a00828c45158#diff-6ec07359a4b7f7c8c986c86549f01867ea071bef35cac64e0bd05302fef2e615), because after creating the model, the user can't change the `max_target_positions` so I saved it in the model.
And finally I added some tests to test this new scenario.